src/emailcorrect.js

```diff
@@ -0,0 +1,70 @@
+const tldMistakes = {
+  com: ['con', 'kom', 'cm', 'cmo', 'cim', 'cpm'],
+  nl: ['ml', 'ln', 'nk', 'bl'],
+  net: ['nt', 'met'],
+};
+
+const hostnameMistakes = {
+  gmail: ['gmai', 'gail', 'gmai', 'gnail', 'mgail', 'gmial', 'mgail', 'gamil'],
+  hotmail: [
+    'htmail', 'otmail', 'hotmal', 'hotmai', 'hotmial', 'hoitmail', 'homail',
+    'hotrmail', 'hotmil', 'hotmaill',
+  ],
+  live: ['lve', 'liv'],
+  yahoo: ['yhoo', 'yaho'],
+};
+
+function last(value = []) {
+  return value[value.length - 1];
+}
+
+function butlast(value = []) {
+  return value.slice(0, value.length - 1);
+}
+
+function disectDomain(domain = '') {
+  // Get TLD and hostname from domain.
+  const parts = domain.split('.');
+  const tld = last(parts);
```

Comment:
Had that before, but even though it doesn't really matter it does mutate the value.